sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/rest/AsyncRestProxy.java

```diff
@@ -176,12 +178,17 @@ static Mono<?> handleBodyReturnType(HttpResponse sourceResponse, Function<byte[]
             // Mono<Flux<ByteBuffer>>
             asyncResult = Mono.just(sourceResponse.getBody());
         } else if (TypeUtil.isTypeOrSubTypeOf(entityType, BinaryData.class)) {
+            HttpHeader httpHeader = sourceResponse.getHeaders().get(HttpHeaderName.fromString("Content-Type"));
```

Comment:
nit: We have `HttpHeaderName.CONTENT_TYPE`

Reply:
Done